pkg/proxy/ipvs/proxier.go

```diff
@@ -562,21 +563,21 @@ func newServiceInfo(port *v1.ServicePort, service *v1.Service, baseInfo *proxy.B
 	return info
 }
 
-// KernelHandler can handle the current installed kernel modules.
+// KernelHandler can handle the current installed kernel modules for ipvs.
 type KernelHandler interface {
+	kernel.Handler
 	GetModules() ([]string, error)
-	GetKernelVersion() (string, error)
 }
 
 // LinuxKernelHandler implements KernelHandler interface.
 type LinuxKernelHandler struct {
-	executor utilexec.Interface
+	kernel.LinuxKernelHandler
 }
 
 // NewLinuxKernelHandler initializes LinuxKernelHandler with exec.
 func NewLinuxKernelHandler() *LinuxKernelHandler {
 	return &LinuxKernelHandler{
-		executor: utilexec.New(),
+		*kernel.NewLinuxKernelHandler(),
```

Comment:
Not a fan of `LinuxKernelHandler` having a field `LinuxKernelHandler`. I think `kernel.NewLinuxKernelHandler` can have a better name like `KernelVersioner` or similar since it is only responsible for getting the kernel version.

Reply:
The kernel NewLinuxKernelHandler may be extended in the future, so I changed the ipvs. LinuxKernelHandler to ipvs. LinuxKernelIPVSHandler